pkg/api/types.go

```diff
@@ -158,13 +158,145 @@ type VolumeSource struct {
 	HostDir *HostDir `json:"hostDir"`
 	// EmptyDir represents a temporary directory that shares a pod's lifetime.
 	EmptyDir *EmptyDir `json:"emptyDir"`
+	// GitRepo represents a git repository at a particular revision.
+	GitRepo *GitRepo `json:"gitRepo"`
+
 	// GCEPersistentDisk represents a GCE Disk resource that is attached to a
 	// kubelet's host machine and then exposed to the pod.
 	GCEPersistentDisk *GCEPersistentDisk `json:"persistentDisk"`
-	// GitRepo represents a git repository at a particular revision.
-	GitRepo *GitRepo `json:"gitRepo"`
+
+	// the named PersistentStorage object this VolumeSource wants to expose to a pod
+	// PersistentStorage must be requested and bound to available storage before it can be used
+	PersistentStorageName string `json:persistentStorageName`
```

Comment:
@smarterclayton need review on the data model and the "high level flow" in the doc.

If this model is in the right direction, i can start hacking a POC.

You'll probably need to click into the file itself for all the structs as I don't see here the couple of structs that live in the file above these lines.